src/vcpkg/base/system.cpp

```diff
@@ -417,17 +417,26 @@ namespace vcpkg
         return ProgramW6432;
     }
 
-    int get_concurrency()
+    unsigned int get_concurrency()
     {
-        static int concurrency = [] {
+        static unsigned int concurrency = [] {
             auto user_defined_concurrency = get_environment_variable("VCPKG_MAX_CONCURRENCY");
             if (user_defined_concurrency)
             {
-                return std::stoi(user_defined_concurrency.value_or_exit(VCPKG_LINE_INFO));
+                try
+                {
+                    int res = std::stoi(user_defined_concurrency.value_or_exit(VCPKG_LINE_INFO));
+                    vcpkg::Checks::check_exit(VCPKG_LINE_INFO, res > 0, "VCPKG_MAX_CONCURRENCY is %d, must be > 0", res);
+                    return static_cast<unsigned int>(res);
+                }
+                catch (std::exception&)
+                {
+                    Checks::msg_exit_with_message(VCPKG_LINE_INFO, msgOptionMustBeInteger, msg::option = "VCPKG_MAX_CONCURRENCY");
```

Comment:
@BillyONeal Should we throw if `VCPKG_MAX_CONCURRENCY` is invalid or should we simply continue with `std::thread::hardware_concurrency()`